### PR TITLE
[FLINK-31922][Connectors/AWS] Port over Kinesis Client configurations for retry and backoff

### PR DESCRIPTION
**Pull request by karubian**
## Purpose of the change

Port over Kinesis Client configurations for retry and backoff

## Verifying this change

- *Added unit tests*

## Significant changes
*(Please check any boxes [x] if the answer is "yes". You can first publish the PR and check them afterwards, for convenience.)*
- [ ] Dependencies have been added or upgraded
- [ ] Public API has been changed (Public API is any class annotated with `@Public(Evolving)`)
- [ ] Serializers have been changed
- [ ] New feature has been introduced
  - If yes, how is this documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/KinesisStreamsSource.java [hunk 1]
```diff
@@ -226,4 +249,37 @@ public UserCodeClassLoader getUserCodeClassLoader() {
...
     }
+
+    private RetryStrategy createExpBackoffRetryStrategy(
+            Duration initialDelay, Duration maxDelay, int maxAttempts) {
+        final BackoffStrategy backoffStrategy =
+                BackoffStrategy.exponentialDelayHalfJitter(initialDelay, maxDelay);
+
+        return StandardRetryStrategy.builder()
+                .backoffStrategy(backoffStrategy)
+                .throttlingBackoffStrategy(backoffStrategy)
```
[the remark this answers is not in the document]
Comment (karubian) [reply flink-connector-aws/flink-connector-aws-kinesis-streams/src/main/java/org/apache/flink/connector/kinesis/source/KinesisStreamsSource.java:259]: We can use but the difference looks marginal. 
Difference between these is the defaults for the attributes below: 
`            .maxAttempts(DefaultRetryStrategy.Standard.MAX_ATTEMPTS)
            .tokenBucketStore(TokenBucketStore
                                  .builder()
                                  .tokenBucketMaxCapacity(DefaultRetryStrategy.Standard.TOKEN_BUCKET_SIZE)
                                  .build())
            .tokenBucketExceptionCost(DefaultRetryStrategy.Standard.DEFAULT_EXCEPTION_TOKEN_COST);`

We are not overriding token bucket store and cost at the moment.

Also, the Standard one is public API and the other one is internal API.

File: flink-connector-aws-base/pom.xml [hunk 2]
```diff
@@ -87,6 +87,12 @@ under the License.
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>kinesis</artifactId>
+            <scope>test</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (karubian) [reply flink-connector-aws-base/pom.xml:94]: This is not needed. Deleted.

File: flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/config/AWSConfigOptions.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription(
+                            "The AWS region of the service (\"us-east-1\" is used if not set).");
+
+    public static final ConfigOption<AWSConfigConstants.CredentialProvider>
+            AWS_CREDENTIALS_PROVIDER_OPTION =
+                    ConfigOptions.key(AWS_CREDENTIALS_PROVIDER)
+                            .enumType(AWSConfigConstants.CredentialProvider.class)
+                            .defaultValue(AWSConfigConstants.CredentialProvider.BASIC)
```
[the remark this answers is not in the document]
Comment (karubian) [reply flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/config/AWSConfigOptions.java:50]: Verified the usage BASIC as default from the old logic. Also discussed offline.

**Comment by hlteoh37** [comment]
Closed as replaced by this PR https://github.com/apache/flink-connector-aws/pull/165

**Pull request closed without merging**